cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+        dest="machines",
+    )
+    dp_upgrade_group.add_argument(
+        "--hostname",
+        "-n",
+        action="append",
+        help="Specify machine hostnames to upgrade.",
+        dest="hostnames",
+    )
+    dp_upgrade_group.add_argument(
+        "--availability-zone",
+        "--az",
+        action="append",
+        help="Specify availability zones to upgrade.",
+        dest="availability_zones",
+    )
+
+    # Arg parser for "cou plan" sub-command
+    plan_parser = subparsers.add_parser(
+        "plan",
+        description="Show the steps for upgrading the cloud to the next release.",
+        help="Show the steps for upgrading the cloud to the next release.",
+        usage="cou plan [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    # Create control-plane and data-plane child commands to plan partial upgrades
+    plan_subparser = plan_parser.add_subparsers(
+        title="Upgrade group",
+        dest="upgrade-group",
+        help="For more information about a upgrade group, run 'cou plan <upgrade-group>' -h.",
+    )
+    plan_subparser.add_parser(
+        "control-plane",
+        description="Show the steps for upgrading the control-plane components.",
+        help="Show the steps for upgrading the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    plan_subparser.add_parser(
+        "data-plane",
+        description="Show the steps for upgrading the data-plane components.",
+        help="Show the steps for upgrading the data-plane components.",
+        usage="cou plan data-plane [options]",
+        parents=[subcommand_common_opts_parser, dp_subparser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    # Arg parser for "cou run" sub-command and set up common options
+    run_args_parser = argparse.ArgumentParser(add_help=False)
+    run_args_parser.add_argument(
         "--interactive",
         help="Run upgrade with prompt.",
         action=argparse.BooleanOptionalAction,
         default=True,
     )
-    parser.add_argument(
-        "-h",
-        "--help",
-        action="help",
-        default=argparse.SUPPRESS,
-        help="Show this help message and exit.",
+    run_args_parser.add_argument(
+        "--i-really-mean-it",
+        help="Run upgrade steps even if they are considered as risky operations (e.g. upgrading "
+        "data-plane nodes with non-empty computes).",
+        action="store_true",
+        default=False,
+    )
+    run_parser = subparsers.add_parser(
+        "run",
+        description="Run the cloud upgrade.",
+        help="Run the cloud upgrade.",
+        usage="cou run [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalisedHelpFormatter,
     )
```

Comment:
it's not clear here if one can run the whole upgrade at once (we must offer this option), i.e. if control-plane and data-plane subparsers are optional

Reply:
Expanded the description to be more informative